CONTRIBUTING.md

```diff
@@ -109,6 +106,66 @@ The contribution process is outlined below:
    maintainer. At this point, a Meta employee will be notified to merge your PR,
    due to tooling limitations.
 
+## Commit Messages
+
+We build Velox for the long-run, and to do so it is crucial that project
+maintainers are able to efficiently inspect and understand project logs.
+
+Commit messages that follow a strict pattern improve maintainability, and allow
+tasks such as summarizing changelogs and identifying API breaking changes to be
+automated. Despite requiring more rigor from authors and reviewers, consistency
+creates productivity in the long term.
+
+In Velox, commit messages are generated based on the input provided to PRs, and
+must follow the [conventional commit
+specification](https://www.conventionalcommits.org/en/v1.0.0/) in the following
+manner:
+
+**PR titles** must follow the pattern:
+
+> \<type\>[optional scope]: \<description\>
+
+where:
+
+* *Type* can be any of the following keywords:
+  * **feat** when new features are being added.
```

Comment:
I feel this one violates 'do-not-abbreviate' guideline that I find useful. Can we drop this altogether and assume that if 'type' is missing then it is a new functionality? Or, change this to 'feature'?

Reply:
Although I also don't like abbreviations in the code, the "feat" naming is part of the conventional commits spec and reasonably well known by folks engaged with other projects. 

I think here, with the tension between abbreviation vs character limit, keeping this as "feat" for consistency may be a fair trade-off, but wonder what others think?